man/systemd-cryptenroll.xml

```diff
@@ -387,6 +387,28 @@
         enrollment to.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--tpm2-external-key=</option><replaceable>PATH</replaceable></term>
+
+        <listitem><para>Enroll an external TPM2 security chip. Expects a path referring to the TPM2 public
+        key in TPM2B_PUBLIC format. This cannot be used with <option>--tpm2-device</option>, and this does
+        not use any TPM for the enrollment. However, once successfully enrolled, the TPM that provided the
+        public key will be able to unlock the LUKS2 volume just as if it was directly enrolled.</para>
+
+        <para>In order to unlock a LUKS2 volume with an externally enrolled TPM2 security chip, use the same
+        configuration in <filename>/etc/crypttab</filename> as would be used for direct TPM
+        enrollment.</para>
+
+        <para>Also see <option>--tpm2-external-location</option>, which is the TPM handle location where the
+        public key must be located to perform the unlocking.</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>--tpm2-external-location=</option><replaceable>LOCATION</replaceable></term>
```

Comment:
"location", is that the right terminology? isn't he right name for this thing "handle"?

Reply:
yeah it's not actually a 'location' but 'handle' seems confusing. i updated it to `--tpm2-handle-index` which i hope is a better way to refer to it? The TPM spec refers to the NV range as 'NV indexes' so maybe it makes sense to refer to all handles as 'handle indexes'. Just referring to it as 'handle' seems confusing since we already have the Tpm2Handle object, and the EAPI's ESYS_TR handle, and then this is the low-level TPM_HANDLE. Hard for me to get a handle on all the handles. :)

I also removed the 'external' part from the index, as specifying the handle index could apply to a local tpm as well.

Technically, we could leave this parameter out compeltely, and just always use the SRK and not give end users the option to use any other TPM handle (index). However it doesn't add any complexity (or at least, not any significant complexity), and I think it could be useful to some people.

Reply:
on thinking about this a bit, i changed it as you suggested to `--tpm2-handle=`. I think at least from the user's perspective, there shouldn't be any confusion.